src/lib.rs

```diff
@@ -101,14 +142,30 @@ fn home_dir_inner() -> Option<PathBuf> {
 /// }
 /// ```
 pub fn cargo_home() -> io::Result<PathBuf> {
-    let cwd = env::current_dir()?;
-    cargo_home_with_cwd(&cwd)
+    cargo_home_from(&OS_ENV)
+}
+
+/// Variant of cargo_home where the environment source is parameterised. This is
+/// specifically to support in-process testing scenarios as environment
+/// variables and user home metadata are normally process global state. See the
+/// OsEnv trait.
+pub fn cargo_home_from(env: &dyn Env) -> io::Result<PathBuf> {
```

Comment:
This could use intra-doc-link for OsEnv trait.

Reply:
How does that look?

Reply:
As defined in this RFC: https://github.com/rust-lang/rfcs/blob/master/text/1946-intra-rustdoc-links.md